samples/fuel_gauge/src/fuel_gauge.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_driver.h>
+#include "nrf_fuel_gauge.h"
+#include "fuel_gauge.h"
+#include <math.h>
+
+#define LOG_MODULE_NAME fuel_gauge
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+static const struct battery_model battery_model = {
+#include "battery_model.inc"
+};
+
+static float max_charge_current;
+static float term_charge_current;
+static int64_t ref_time;
+
+static int read_sensors(npmx_instance_t *const p_pm, float *voltage, float *current, float *temp)
+{
+	npmx_adc_t *adc_instance = npmx_adc_get(p_pm, 0);
+	npmx_adc_meas_all_t meas;
+	npmx_adc_ibat_meas_status_t status;
+
+	if (npmx_adc_meas_all_get(adc_instance, &meas) != NPMX_SUCCESS) {
+		LOG_ERR("Reading ADC measurements failed.");
+		return -EIO;
+	}
+
+	if (npmx_adc_ibat_meas_status_get(adc_instance, &status) != NPMX_SUCCESS) {
+		LOG_ERR("Reading IBAT status failed.");
+		return -EIO;
+	}
+
+	if (status.invalid == true) {
+		LOG_ERR("IBAT measurement is invalid.");
+		return -EINVAL;
+	}
```

Comment:
This might not be needed as `npmx_adc_ibat_meas_status_get` will return error when measurement is invalid, unless it will be changed back in https://github.com/nrfconnect/npmx/pull/112